### PR TITLE
Fixed Issue 3649 by adding 'pip install...' to ImportErrors

### LINKED ISSUE #3649: Imports of optional dependencies should happen at runtime
Across terra, there are several imports of optional dependencies (packages other than those in requirements.txt), which happen at top-level (not within another function definition.) When a user hits an error message about a missing optional dependency, just running `pip install foo` won't be sufficient to use the package, they'll need to restart their kernel. With a runtime, import, the package would be available in their current session.

e.g. from `dag_visualization.py`
```
try:
    from PIL import Image
    HAS_PIL = True
except ImportError:
    HAS_PIL = False

def draw(...):
   ...
        if not HAS_PIL:
            raise ImportError(
                "dag_drawer requires pillow for drawing in jupyter directly. "
                "Run 'pip install pillow'.")
```

Searching for `except ImportError` and narrowing down to cases that happen at top-level is probably the best way to find occurrences.

### PR DESCRIPTION
**Pull request by michelle4654**
### Summary

~Fixes~ Related to #3649. Adding 'pip install [package]' whenever there's an ImportError. This will help give directions to users when there's an error.

### CODE REVIEW

File: qiskit/tools/jupyter/backend_monitor.py [hunk 1]
```diff
@@ -30,7 +30,8 @@
     # pylint: disable=import-error
     from qiskit.providers.ibmq import IBMQBackend
 except ImportError:
-    pass
+    raise ImportError('Could not import the IMBQ provider from the qiskit-ibmq-provider'
```
Comment (kdk) [inline qiskit/tools/jupyter/backend_monitor.py]: In this case, the `ImportError` is not necessary. (We check HAS_IBMQ in jupyter/__init__.py before we register _backend_monitor.)

File: qiskit/test/decorators.py [hunk 2]
```diff
@@ -90,7 +90,9 @@ def _get_credentials():
                                                        discover_credentials)
     except ImportError:
         raise unittest.SkipTest('qiskit-ibmq-provider could not be found, '
-                                'and is required for executing online tests.')
+                                'and is required for executing online tests.'
```
Comment (kdk) [inline qiskit/test/decorators.py]:
```suggestion
                                'and is required for executing online tests. '
```
When splitting strings across multiple lines (here and a few other places), there should be a trailing space.

**Comment by michelle4654** [comment]
Hi! I think I added all the trailing spaces and updated the backend_monitor appropriately!

File: qiskit/test/decorators.py [hunk 3]
```diff
@@ -90,7 +90,9 @@ def _get_credentials():
                                                        discover_credentials)
     except ImportError:
         raise unittest.SkipTest('qiskit-ibmq-provider could not be found, '
-                                'and is required for executing online tests.')
+                                'and is required for executing online tests. '
+                                'Install iskit-ibmq-provider or check your '
+                                'installation. ')
```
Comment (kdk) [inline qiskit/test/decorators.py]:
```suggestion
                                'installation.')
```

**Comment by michelle4654** [comment]
Oops! Yeah, that makes sense -- thought I just didn't understand something about the convention, sorry!

File: qiskit/test/decorators.py [hunk 4]
```diff
@@ -90,7 +90,9 @@ def _get_credentials():
                                                        discover_credentials)
     except ImportError:
         raise unittest.SkipTest('qiskit-ibmq-provider could not be found, '
-                                'and is required for executing online tests.')
+                                'and is required for executing online tests. '
+                                'Install iskit-ibmq-provider or check your '
```
Comment (maddy-tod) [inline qiskit/test/decorators.py]: Could you update this message to say how to install the IBMQ provider, like how the other messages are? `"pip install qiskit-ibmq-provider"`

**Comment by kdk** [comment]
Thanks @michelle4654 !

**Comment by michelle4654** [comment]
Thank you for all the help!!

**Pull request merged**